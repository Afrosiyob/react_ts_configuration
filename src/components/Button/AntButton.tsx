/* eslint-disable react/jsx-props-no-spreading */
import React, { ReactElement } from "react";
import { Button } from "antd";

import { Types } from ".";

function AntButton({
  onClick,
  children = "Button",
  ...props
}: Types.IButton): ReactElement {
  return (
    <Button data-testid="ANT_BUTTON_ID" onClick={onClick} {...props}>
      {children}
    </Button>
  );
}

export default AntButton;
