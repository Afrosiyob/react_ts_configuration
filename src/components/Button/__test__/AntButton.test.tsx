import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import AntButton from "../AntButton";

describe("Test for Component 'AntButton'", () => {
  it("should render", () => {
    render(<AntButton> Button </AntButton>);

    const component = screen.getByTestId("ANT_BUTTON_ID");

    expect(component).toBeInTheDocument();
  });

  it("should work props", () => {
    const testRenderer = renderer.create(
      <AntButton size="middle" disabled={true}>
        Button
      </AntButton>
    );
    const testInstance = testRenderer.root;

    expect(testInstance.props.size).toBe("middle");
    expect(testInstance.props.disabled).toBe(true);
  });

  // it("should work action", () => {
  //   const onClick = jest.fn();

  //   render(<AntButton> Button </AntButton>);
  //   const component = screen.getByTestId("ANT_BUTTON_ID");

  //   fireEvent.click(component);

  //   expect(onClick).toHaveBeenCalled();
  // });
});

describe("Snapshot for component 'AntButton'", () => {
  it("should matches DOM", () => {
    const tree = renderer.create(<AntButton> Button </AntButton>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
