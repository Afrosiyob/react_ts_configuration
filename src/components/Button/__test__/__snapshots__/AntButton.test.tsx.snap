// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for component 'AntButton' should matches DOM 1`] = `
<button
  className="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-default"
  data-testid="ANT_BUTTON_ID"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span>
     Button 
  </span>
</button>
`;
